### PR TITLE
Update JS dependencies

### PR DESCRIPTION
**Pull request by silverwind**
- Update all JS deps
- Regenerate SVGs
- Add new eslint rules, fix issues
- Tested Mermaid, Swagger, Vue, Webpack, Citation

### CODE REVIEW

**Review by delvh — APPROVED** [review]

**Comment by delvh** [comment]
… Your PR on khroma was denied… :smiling_face_with_tear:

**Comment by silverwind** [comment]
Stubbornly, yes. Hundrets of package manage to have the license field, but there's always that one stupid exception.

**Comment by fabiospampinato** [comment]
It's a well motivated reason (https://github.com/fabiospampinato/khroma/pull/33#issuecomment-1321250887), those tools are unreliable almost by definition. But sure, I can understand the frustration.

**Comment by silverwind** [comment]
Of note is that `license-checker-webpack-plugin` is primarly there to extract the licenses, the checking functionality is more of a side effect. If in the future we find a way to only extract licenses without checking, I'm happy to switch over to that. Might be necessary anyways once we migrate off webpack.

**Comment by fabiospampinato** [comment]
>  If in the future we find a way to only extract licenses without checking, I'm happy to switch over to that. Might be necessary anyways once we migrate off webpack.

fwiw I'd like to eventually build a standalone little library that given the current package extracts all of its NPM dependencies, detects their repositories, their readmes, their license files, and their license identifier (like just "MIT" rather than the whole string). If you might consider using this and there's some other feature that you'd need just let me know 👍

**Review by a1012112796 — APPROVED** [review]

File: webpack.config.js [hunk 1]
```diff
@@ -227,10 +227,11 @@ export default {
         }).join('\n');
       },
       override: {
-        'jquery.are-you-sure@*': {licenseName: 'MIT'},
+        'jquery.are-you-sure@*': {licenseName: 'MIT'}, // https://github.com/codedance/jquery.AreYouSure/pull/147
+        'khroma@*': {licenseName: 'MIT'}, // https://github.com/fabiospampinato/khroma/pull/33
       },
       emitError: true,
-      allow: '(Apache-2.0 OR BSD-2-Clause OR BSD-3-Clause OR MIT OR ISC OR CPAL-1.0)',
+      allow: '(Apache-2.0 OR BSD-2-Clause OR BSD-3-Clause OR MIT OR ISC OR CPAL-1.0 OR Unlicense)',
```
Comment (lunny) [inline webpack.config.js:234]: Should we allow `Unlicense` packages come in?

Comment (silverwind) [reply webpack.config.js:234]: I see no issue, it's [essentially public domain](https://en.wikipedia.org/wiki/Unlicense). If we don't we have to drop mermaid. It's coming from `robust-predicates`:

```
gitea@
└─┬ mermaid@9.2.2
  └─┬ d3@7.6.1
    └─┬ d3-delaunay@6.0.2
      └─┬ delaunator@5.0.0
        └── robust-predicates@3.0.1
```

**Review by delvh — APPROVED** [review]

**Comment by lunny** [comment]
make L-G-T-M work

**Pull request merged**